subsys/bluetooth/host/conn.c

```diff
@@ -1502,11 +1502,21 @@ int bt_conn_disconnect(struct bt_conn *conn, uint8_t reason)
 		if (conn->type == BT_CONN_TYPE_BR) {
 			return bt_hci_connect_br_cancel(conn);
 		}
+		else
 #endif /* CONFIG_BT_BREDR */
-
-		if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
-			k_work_cancel_delayable(&conn->deferred_work);
-			return bt_le_create_conn_cancel();
+		if (conn->type == BT_CONN_TYPE_LE) {
+			if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
+				k_work_cancel_delayable(&conn->deferred_work);
+				return bt_le_create_conn_cancel();
+			}
+		}
+#if defined(CONFIG_BT_ISO)
+		else if (conn->type == BT_CONN_TYPE_ISO) {
+			return conn_disconnect(conn, reason);
+		}
+#endif
+		else {
+			__ASSERT(false, "Invalid conn type %u", conn->type);
```

Comment:
Doesn't `bt_conn_disconnect` also work on BREDR conns?

Reply:
There we call `bt_hci_connect_br_cancel()`